superset/config.py

```diff
@@ -1107,6 +1107,8 @@ def CSV_TO_HIVE_UPLOAD_DIRECTORY_FUNC(  # pylint: disable=invalid-name
 
 # Interval between consecutive polls when using Hive Engine
 HIVE_POLL_INTERVAL = int(timedelta(seconds=5).total_seconds())
+# Interval between consecutive polls when using Impala Engine
+IMPALA_POLL_INTERVAL = int(timedelta(seconds=5).total_seconds())
```

Comment:
To make sure we don't clutter `config.py` with too many disparate config keys, maybe we should remove these `*_POLL_INTERVAL` keys and refactor this to something like
```python
DB_POLL_INTERVAL_SECONDS: Dict[str, int] = {}
```
This could be used to specify these per engine name in your `superset_config.py` (here I'd be overriding polling to 1 seconds for Hive):
```python
DB_POLL_INTERVAL_SECONDS = {
    "hive": int(timedelta(seconds=1).total_seconds()),
}
```
I know it's a breaking change, so we can probably fall back to `HIVE_POLL_INTERVAL` in the hive spec for now. So maybe change the following line https://github.com/apache/superset/blob/01671b9d1b3a15c264bcfb9eced1776c70e293b5/superset/db_engine_specs/hive.py#L378 to something like this (see how I've moved the default to the engine spec, rather than having it in `config.py`):

```python
if sleep_interval := current_app.config.get("HIVE_POLL_INTERVAL"):
    logger.warning("HIVE_POLL_INTERVAL is deprecated and will be removed in 3.0. Please use DB_POLL_INTERVAL instead")
else:
    sleep_interval = current_app.config["DB_POLL_INTERVAL_SECONDS"].get(cls.engine, 5)

time.sleep(sleep_interval)
```

Reply:
This is a good idea. I will try it locally and then submit the code